### PR TITLE
Improve documentaion for installing Bazel on Windows

### PR DESCRIPTION
**Pull request by meteorcloudy**
Improvements:

- Moved optional steps to the new "Best practices" section
- Addressed comments in #11318

Fixes: https://github.com/bazelbuild/bazel/issues/11318

### CODE REVIEW

File: site/docs/install-windows.md [hunk 1]
```diff
@@ -9,19 +9,14 @@ title: Installing Bazel on Windows
 
 ### Step 1: Check your system
 
-Recommended: 64 bit Windows 10, version 1703 (Creators Update) or newer, enable "Developer Mode".
-
-<!-- Developer mode: for symlink support. -->
+Recommended: 64 bit Windows 10, version 1703 (Creators Update) or newer
```
Comment (laurentlb) [inline site/docs/install-windows.md:12]: How do I check the Windows version?

File: site/docs/install-windows.md [hunk 2]
```diff
@@ -41,38 +34,63 @@ Alternatively you can:
...
-### Step 4 (optional): Configure output directories
+### Step 4: Set up your environment
+
+It's a good idea to rename the Bazel binary to `bazel.exe` and add it to your default paths.
+
+```batch
+set PATH=%PATH%;<path to the Bazel binary>
+```
+
+You can also change your system `PATH` environment variable to make it permanent.
```
Comment (laurentlb) [inline site/docs/install-windows.md:45]: Can we add a link that explains how to do that?

File: site/docs/install-windows.md [hunk 3]
```diff
@@ -41,38 +34,63 @@ Alternatively you can:
...
-its default values.**
+### Step 5: Done
 
-By default, Bazel writes to two directories:
+**You have successfully installed Bazel.**
+To check the installation is correct, try to run:
+```batch
+bazel version
+```
+
```
Comment (jin) [inline site/docs/install-windows.md:54]: "Next, you can check out some best practices, troubleshooting tips and tutorials available below." and remove the list?

File: site/docs/install-windows.md [hunk 4]
```diff
@@ -41,38 +34,63 @@ Alternatively you can:
 *   [Install Bazel from Scoop](#using-scoop)
 *   [Build Bazel from source](install-compile-source.html)
 
-### Step 4 (optional): Configure output directories
+### Step 4: Set up your environment
+
+It's a good idea to rename the Bazel binary to `bazel.exe` and add it to your default paths.
```
Comment (jin) [inline site/docs/install-windows.md:39]:
```suggestion
To make Bazel easily accessible from command prompts or PowerShell by default, you can rename the Bazel binary to `bazel.exe` and add it to your default paths.
```

File: site/docs/install-windows.md (hunk 2 above)
Comment (meteorcloudy) [reply site/docs/install-windows.md:45]: Done

File: site/docs/install-windows.md (hunk 1 above)
Comment (meteorcloudy) [reply site/docs/install-windows.md:12]: Instruction added

File: site/docs/install-windows.md [hunk 5]
```diff
@@ -41,38 +34,63 @@ Alternatively you can:
...
+```
+startup --windows_enable_symlinks
+build --enable_runfiles
+```
 
-    You can tell Bazel not to create these junctions with `--symlink_prefix=/`.
+**Note that**: Creating symlink on Windows is an expensive operation, `--enable_runfiles` sometimes means creating a large amount of file symlinks. Only enable this feature when you are sure you need it.
 
-### Step 5 (optional): Install compilers and language runtimes
+<!-- TODO(pcloudy): Write a doc about runfiles library and add a link to it here -->
```
[the remark this answers is not in the document]
Comment (meteorcloudy) [reply site/docs/install-windows.md]: Done.
There is already a bug for that, I added it here in comment.

File: site/docs/install-windows.md [hunk 6]
```diff
@@ -41,38 +34,63 @@ Alternatively you can:
...
+```
+startup --output_user_root=C:/tmp
+```
 
-    This is where Bazel writes all output files. By default, this is a subdirectory of the "output
-    user root".
+### Enable symlink support
 
-By default, Bazel also writes in the workspace directory:
+Some features require Bazel to create file symlink on Windows, you can allow Bazel to do that by enabling [Developer Mode](https://docs.microsoft.com/en-us/windows/uwp/get-started/enable-your-device-for-development) on Windows.
```
[the remark this answers is not in the document]
Comment (meteorcloudy) [reply site/docs/install-windows.md]: The features are enabled by the two flags listed below. I rephrased this to make it more clear. Thanks!

File: site/docs/install-windows.md (hunk 3 above)
Comment (meteorcloudy) [reply site/docs/install-windows.md:59]: I moved the best practices section to a windows.md. Here I think it's useful to give a list of links to additional useful infromation.

**Comment by meteorcloudy** [comment]
Thanks for the review! I made more improvements to Windows doc, PTAL!

**Comment by jin** [comment]
Thanks!

**Comment by limdor** [comment]
I just came across this PR when I was trying to figure out when it happened that the Windows documentation improved so much.
This is really great work.

**Comment by meteorcloudy** [comment]
@limdor Thanks!! Glad to help!

**Pull request closed without merging**